repl/src/CodeMirror6.jsx

```diff
@@ -19,60 +18,29 @@ const highlightField = StateField.define({
   },
   update(highlights, tr) {
     try {
-      highlights = highlights.map(tr.changes);
       for (let e of tr.effects) {
-        if (e.is(addHighlight)) {
-          highlights = highlights.update({
-            add: [highlightMark.range(e.value.from, e.value.to)],
-          });
-        }
-        if (e.is(removeHighlight)) {
-          highlights = highlights.update({
-            filter: (f, t, value) => {
-              if (f === e.value.from && t === e.value.to) {
-                return false;
-              }
-              return true;
-              // console.log('filter', f,t,value, e.value.from, e.value.to);
-            },
-          });
+        if (e.is(setHighlights)) {
```

Comment:
that looks much simpler than what I fabricated :D didn't know you could just set the Decorations directly